src/test/java/com/GeorgesServer/app/EstablishesConnectionTest.java

```diff
@@ -1,31 +1,56 @@
 package com.GeorgesServer.app;
 
-
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-import java.io.IOException;
+import java.io.*;
 import java.net.ServerSocket;
 import java.net.Socket;
 
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 public class EstablishesConnectionTest {
     private EstablishesConnection subject;
-    private int validPortNumber = 5000;
-    private String invalidPortNumber = "invalid";
-    private ServerSocket mockedServerAccept;
-    private Socket mockedSocket;
+    private SocketFactory mockedSocketFactory;
+    private ServerSocket mockedServerSocket;
+    private InputStream mockedInputStream;
+    private InputStreamReader mockedInputStreamReader;
+    private BufferedReader mockedBufferedReader;
+    private OutputStream mockedOutputStream;
+    private OutputStreamWriter mockedOutputStreamWriter;
+    private Socket mockedClientSocket;
 
     @BeforeEach
     void setUp () {
-        subject = new EstablishesConnection();
-        mockedServerAccept = mock(ServerSocket.class);
-        mockedSocket = mock(Socket.class);
+        mockedSocketFactory = mock(SocketFactory.class);
+        mockedServerSocket = mock(ServerSocket.class);
+        mockedClientSocket = mock(Socket.class);
+        mockedInputStream = mock(InputStream.class);
+        mockedInputStreamReader = mock(InputStreamReader.class);
+        mockedOutputStream = mock(OutputStream.class);
+        mockedOutputStreamWriter = mock(OutputStreamWriter.class);
+        mockedBufferedReader = mock(BufferedReader.class);
+        subject = new EstablishesConnection(mockedSocketFactory);
     }
 
     @Test
-    void establishesConnectionReturnsAConnectionsObject() throws IOException { /* TODO */ }
+    void establishesConnectionCreatesAnInputAndOutputConnection() {
+        int validPortNumber = 6000;
+        when(mockedSocketFactory.createSocket(validPortNumber)).thenReturn(mockedServerSocket);
```

Comment:
none of these mocks are necessary.

Reply:
Or actually they might be but the fact that they are is obfuscated by the multiple responsibilities of the socketFactory